src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/PackageUpdaterStepExtensions.cs

```diff
@@ -14,16 +14,19 @@ public static class PackageUpdaterStepExtensions
 
         public static void AddPackageUpdaterStep(this IExtensionServiceCollection services)
         {
+
             if (services is null)
             {
                 throw new System.ArgumentNullException(nameof(services));
             }
 
             services.Services.AddUpgradeStep<PackageUpdaterPreTFMStep>();
             services.Services.AddUpgradeStep<PackageUpdaterStep>();
+            services.Services.AddSingleton<IDependencyAnalyzerRunner, DependencyAnalyzerRunner>();
```

Comment:
@twsouthwick Is this the right way of adding dependencies with the new abstraction model you just merged?

Reply:
yup. Although transient may not be a bad choice here either like Mike said